samcli/local/apigw/local_apigw_service.py

```diff
@@ -234,11 +235,47 @@ def _should_base64_decode_body(binary_types, flask_request, lamba_response_heade
         True if the body from the request should be converted to binary, otherwise false
 
         """
-        best_match_mimetype = flask_request.accept_mimetypes.best_match([lamba_response_headers["Content-Type"]])
+
+        # Get the first part of the content-type header, to allow for extras such as text/html; charset=utf-8
+        content_type = lamba_response_headers['Content-Type'].split(";", 1)[0]
+        best_match_mimetype = flask_request.accept_mimetypes.best_match([content_type])
         is_best_match_in_binary_types = best_match_mimetype in binary_types or '*/*' in binary_types
 
         return best_match_mimetype and is_best_match_in_binary_types and is_base_64_encoded
 
+    @staticmethod
+    def _merge_response_headers(headers, multi_headers):
+        """
+        # Merge multiValueHeaders headers with headers
+        # Convert into CSV for Flask compatibility
+        # If you specify values for both headers and multiValueHeaders, API Gateway merges them into a single list.
+        # If the same key-value pair is specified in both, only the values from multiValueHeaders will
+        # appear in the merged list.
+
+        Parameters
+        ----------
+        headers (dict)
+            Headers map from the lambda_response_headers
+        multi_headers (dict)
+            multiValueHeaders map from the lambda_response_headers
+
+        Returns
+        -------
+        Merged list in accordance to the AWS documentation within a CaseInsensitiveDict
+
+        """
+
+        # Merge dictionary
+        h = (headers or {}).copy()
+        h.update(multi_headers or {})
+
+        # Now we have single headers map with Strings and Lists, convert Lists to CSV Strings for Flask
+        for _, s in enumerate(h):
```

Comment:
You should just iterate over the `multi_headers`. The value of each header will be a list, so you won't need the `isinstance` check.